src/js/index.js

```diff
@@ -1,128 +1,98 @@
+      case 'bar':
+        this.highchart = createChart.bar( chartOptions );
+        break;
+      case 'tile_map':
+        this.highchart = createChart.tileMap( chartOptions );
+        break;
+      default:
     }
-    urls[url].push( chart );
   }
 
-  for ( var key in urls ) {
-
-    loadSource( key, function( key, data ) {
-      for ( var x = 0; x < urls[key].length; x++ ) {
-        var chart = urls[key][x],
-            type = chart.getAttribute( 'data-chart-type' ),
-            group = chart.getAttribute( 'data-chart-metadata' ),
-            color = chart.getAttribute( 'data-chart-color' );
-
-        // Ensure undefined attributes aren't cast as a string.
-        group = group === 'undefined' ? undefined : group;
-
-
-        var properties = {
-          type: type,
-          selector: chart,
-          color: color
-        };
+  update( newOptions ) {
+    // Merge the old chart options with the new ones
+    Object.assign( this.chartOptions, newOptions );
+    // If the source wasn't changed, we don't need to fetch new data and can
+    // immediately redraw the chart
+    if ( !newOptions.source ) {
+      return this.highchart.update( this.chartOptions );
+    }
+    // Otherwise fetch the data and redraw once it arrives
+    this.highchart.chart.showLoading( ' ' );
+    ajax( this.chartOptions.source ).then( data => {
+      this.chartOptions.data = data;
+      this.highchart.update( this.chartOptions );
+    } );
+    return this;
+  }
 
-        if ( type === 'line' ) {
-          properties.data = process.originations( data, group );
+}
 
-          if ( typeof properties.data === 'object' ) {
-            createChart.line( properties );
-          } else {
-            chart.setAttribute( 'data-chart-error', errorStrings[properties.data] );
-            console.log( errorStrings[properties.data] );
-          }
-        }
+function _createChart( opts ) {
+  return new Chart( opts );
+}
 
-        if ( type === 'bar' ) {
-          properties.data = process.yoy( data, group );
-          if ( typeof properties.data === 'object' ) {
-            createChart.bar( properties );
-          } else {
-            chart.setAttribute( 'data-chart-error', errorStrings[properties.data] );
-            console.log( errorStrings[properties.data] );
-          }
-        }
+function _createCharts() {
+  let elements = document.querySelectorAll( '.cfpb-chart' );
+  let charts = [];
 
-        if ( type === 'tile_map' ) {
-          properties.data = process.map( data, group );
-          if ( typeof properties.data === 'object' ) {
-            createChart.map( properties );
-          } else {
-            chart.setAttribute( 'data-chart-error', errorStrings[properties.data] );
-            console.log( errorStrings[properties.data] );
-          }
-        }
+  // Ignore divs with a `data-chart-ignore` data attribute
+  for ( let i = 0; i < elements.length; ++i ) {
+    if ( !elements[i].getAttribute( 'data-chart-ignore' ) ) {
```

Comment:
In order for the new line comparison chart to look like our other charts, I had to give it the `cfpb-chart` class because that's how our stylesheet works. Buuuut by doing that the JS tries to render a chart in the div because it gets picked up by `document.querySelectorAll( '.cfpb-chart' )`. So I created a new attribute called `data-chart-ignore` that can be added to divs if they wish to be ignored by the automatic chart instantiation process.

There's probably a better way of handling this in the future. Maybe we should use different selectors for the JS and CSS? Something to think about.